src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,189 +1,376 @@
+                        val className = group("className")
+                        val level = group("level").toInt()
+                        val color = getColor(level)
+                        if (config.coloredClassLevel) toolTip[index] = " §b$playerName§f: §e$className $color$level"
+                        classNames.remove(className)
+                    }
+                }
+                if (config.showMissingClasses && dungeonFloorPattern.matches(
+                        stack.getLore().firstOrNull()?.removeColor()
+                    )
+                ) {
+                    if (classNames.contains(selectedClass)) classNames[classNames.indexOf(selectedClass)] =
+                        "§a${selectedClass}§7"
+                    toolTip.add("")
+                    toolTip.add("§cMissing: §7" + StringUtils.createCommaSeparatedList(classNames))
+                }
+                if (toolTip.isNotEmpty()) toolTipMap[slot] = toolTip
             }
         }
     }
 
     @SubscribeEvent
-    fun onItemTooltip(event: LorenzToolTipEvent) {
-        if (!LorenzUtils.inSkyBlock) return
-
-        val chestName = InventoryUtils.openInventoryName()
-        if (chestName != "Party Finder") return
-
-        val stack = event.itemStack
-
-        val classNames = mutableListOf("Healer", "Mage", "Berserk", "Archer", "Tank")
-        for ((index, line) in stack.getLore().withIndex()) {
-            classLevelPattern.matchMatcher(line) {
-                val playerName = group("playerName")
-                val className = group("className")
-                val level = group("level").toInt()
-                val color = getColor(level)
-                if (config.coloredClassLevel) event.toolTip[index + 1] = " §b$playerName§f: §e$className $color$level"
-                classNames.remove(className)
+    fun onToolTipRender(event: LorenzToolTipEvent) {
+        if (!isEnabled()) return
+        if (!inInventory) return
+        val toolTip = toolTipMap[event.slot.slotIndex]
+        if (toolTip.isNullOrEmpty()) return
+        // TODO @Thunderblade73 fix that to "event.toolTip = toolTip"
+        val oldToolTip = event.toolTip
+        for ((index, line) in toolTip.withIndex()) {
+            if (index >= event.toolTip.size - 1) {
+                event.toolTip.add(line)
+                continue
             }
+            if (oldToolTip[index] != line) event.toolTip[index + 1] = line
         }
-        if (!config.showMissingClasses) return
-        if (stack.getLore().firstOrNull()?.removeColor()?.startsWith("Dungeon:") == false) return
-        if (classNames.contains(selectedClass)) selectedClass = "§a${selectedClass}§7"
-        event.toolTip.add("")
-        event.toolTip.add("§cMissing: §7" + createCommaSeparatedList(classNames))
     }
 
     @SubscribeEvent
-    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
-        event.move(2, "dungeon.partyFinderColoredClassLevel", "dungeon.partyFinder.coloredClassLevel")
```

Comment:
while the 2nd round of config migration was forever ago I don't think we should be removing it here, this might of been an accident

Reply:
yea agreed I removed it by accident